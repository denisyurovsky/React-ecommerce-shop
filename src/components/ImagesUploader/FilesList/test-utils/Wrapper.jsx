import PropTypes from 'prop-types';
import React from 'react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';

import { FilesList } from '../FilesList';

const Wrapper = ({ isLoad, files, setCroppedImage, handleDelete }) => (
  <DndProvider backend={HTML5Backend}>
    <FilesList
      handleImageDelete={handleDelete}
      filesList={files}
      setCroppedImage={setCroppedImage}
      isImagesLoad={isLoad}
    />
  </DndProvider>
);

Wrapper.defaultProps = {
  isLoad: false,
  files: [],
  setCroppedImage: () => {},
  handleDelete: () => {},
};

Wrapper.propTypes = {
  isLoad: PropTypes.bool,
  files: PropTypes.arrayOf(PropTypes.object),
  setCroppedImage: PropTypes.func,
  handleDelete: PropTypes.func,
};

export default Wrapper;
