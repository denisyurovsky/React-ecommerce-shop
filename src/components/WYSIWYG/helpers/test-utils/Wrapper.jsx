import { EditorState } from 'draft-js';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

import TextEditor from '../../TextEditor';

const Wrapper = ({ error, helperText }) => {
  const [editorState, setEditorState] = useState(EditorState.createEmpty());

  return (
    <TextEditor
      error={error}
      helperText={helperText}
      editorState={editorState}
      setEditorState={setEditorState}
    />
  );
};

Wrapper.defaultProps = {
  error: false,
  helperText: null,
};

Wrapper.propTypes = {
  error: PropTypes.bool,
  helperText: PropTypes.string,
};

export default Wrapper;
