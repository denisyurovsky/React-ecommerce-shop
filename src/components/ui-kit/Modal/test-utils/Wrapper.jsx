import { Button } from '@mui/material';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

import { MODAL_SIZES } from '../../../../constants/constants';
import Modal from '../Modal';

const { NORMAL, SMALL } = MODAL_SIZES;

const Wrapper = ({
  title,
  actionButtonLabel,
  cancelButtonLabel,
  onConfirm,
  size,
}) => {
  const [isOpen, setIsOpen] = useState(false);

  const handleOpen = () => setIsOpen(true);
  const handleClose = () => setIsOpen(false);

  return (
    <>
      <Button onClick={handleOpen}>Modal</Button>
      <Modal
        isOpen={isOpen}
        onClose={handleClose}
        title={title}
        actionButtonLabel={actionButtonLabel}
        cancelButtonLabel={cancelButtonLabel}
        onConfirm={onConfirm}
        size={size}
      >
        <div>Content</div>
      </Modal>
    </>
  );
};

Wrapper.defaultProps = {
  title: null,
  actionButtonLabel: null,
  cancelButtonLabel: null,
  onConfirm: null,
};

Wrapper.propTypes = {
  title: PropTypes.string,
  actionButtonLabel: PropTypes.string,
  cancelButtonLabel: PropTypes.string,
  onConfirm: PropTypes.func,
  size: PropTypes.oneOf([NORMAL, SMALL]),
};

export default Wrapper;
