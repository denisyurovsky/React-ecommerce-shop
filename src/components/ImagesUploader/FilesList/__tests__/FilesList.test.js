import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import {
  croppedWithState,
  unCroppedWithState,
  croppingWithState,
} from '../../../../test-utils/images/images';
import Wrapper from '../test-utils/Wrapper';

describe('FilesList component', () => {
  it('renders empty component without data', () => {
    const { asFragment } = render(<Wrapper />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component with data', () => {
    const { asFragment } = render(<Wrapper files={[unCroppedWithState]} />);
    const imgItem = screen.getByTestId('imgItem');

    expect(screen.getByTestId('imgItem')).toBeInTheDocument();
    expect(imgItem).toHaveTextContent('unCropped.png');
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render spinner on load', () => {
    render(<Wrapper files={[unCroppedWithState]} isLoad={true} />);

    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });

  describe('callback functionality', () => {
    const deleteCallback = jest.fn();
    const changeCallback = jest.fn();

    beforeEach(() => {
      render(
        <Wrapper
          files={[croppedWithState, unCroppedWithState]}
          handleDelete={deleteCallback}
          setCroppedImage={changeCallback}
        />
      );
    });

    it('should be able to set images for crop', () => {
      userEvent.click(screen.getByText('unCropped.png', { exact: false }));
      expect(changeCallback).toHaveBeenCalledWith(croppedWithState.file);
    });

    it('should be able to delete images', () => {
      userEvent.click(screen.getAllByTestId('DeleteIcon')[0]);
      expect(deleteCallback).toHaveBeenCalledTimes(1);
    });
  });

  it('should style cropped images', () => {
    render(<Wrapper files={[croppedWithState]} />);

    expect(screen.getByTestId('CheckCircleIcon')).toBeInTheDocument();
  });

  it('should style cropping images', () => {
    render(<Wrapper files={[croppingWithState]} />);

    expect(screen.getByTestId('CropIcon')).toBeInTheDocument();
  });
});
