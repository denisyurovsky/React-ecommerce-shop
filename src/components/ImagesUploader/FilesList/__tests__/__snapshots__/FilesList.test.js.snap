// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesList component renders component with data 1`] = `
<DocumentFragment>
  <div
    class="list MuiBox-root css-0"
  >
    <div
      class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
    >
      Selected images:
    </div>
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
        data-testid="imgItem"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth MuiButtonBase-root btn css-10vjzfb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiListItemAvatar-root previewWrapper css-1e9lk82-MuiListItemAvatar-root"
          >
            <img
              alt="unCropped.png"
              class="preview"
            />
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              unCropped.png - 0KB
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
        >
          <button
            aria-label="delete"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            data-testid="delete"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`FilesList component renders empty component without data 1`] = `<DocumentFragment />`;
